pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +383,108 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+def _find_diag_from_eye_mul(node):
```

Comment:
Why did you separate this into a separate function?

Reply:
this will be used in other diag-related rewrites, like inv(diag(x)) -> diag(1/x)

Reply:
alright. is "potential_mul_input" a better name for what u are saying? that way we do have an idea of why the check directly starts from finding an Elemwise Mul Op

Reply:
Yes, but pass already the variable that here was retrived by node.inputs[0] that part doesn't make sense